compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -284,6 +327,10 @@ pub fn create_wrapper_file(
             file.section_mut(section).flags =
                 SectionFlags::Elf { sh_flags: elf::SHF_EXCLUDE as u64 };
         }
+        BinaryFormat::Xcoff => {
```

Comment:
Is it missing a `C_INFO` symbol to reference the data in `.info` section?
As @bjorn3 said, we don't need to preserve the metadata here, so it doesn't need a data symbol and a csect here compared to `create_compressed_metadata_file_for_xcoff`.